File: src/api/MoRS.ManagementSystem.Application/DTOs/MalfunctionReport/CreateMalfunctionReportRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MoRS.ManagementSystem.Application.DTOs.MalfunctionReport;

public class CreateMalfunctionReportRequest
{
    [Required]
    [StringLength(1000, MinimumLength = 10)]
    public required string Description { get; set; }

    [Required]
    [Range(1, int.MaxValue)]
    public int RoomId { get; set; }

    [Required]
    [Range(1, int.MaxValue)]
    public int ReportedByUserId { get; set; }
}

File: src/mobile/lib/models/malfunction_report/create_malfunction_report_request.dart
class CreateMalfunctionReportRequest {
  final String description;
  final int roomId;
  final int reportedByUserId;

  CreateMalfunctionReportRequest({
    required this.description,
    required this.roomId,
    required this.reportedByUserId,
  });

  Map<String, dynamic> toJson() {
    return {
      'description': description,
      'roomId': roomId,
      'reportedByUserId': reportedByUserId,
    };
  }
}
